lib/osf-components/addon/components/search-page/filter-facet/component.ts

```diff
@@ -55,33 +71,90 @@ export default class FilterFacet extends Component<FilterFacetArgs> {
         this.selectedProperty = property;
     }
 
+    @action
+    openSeeMoreModal() {
+        this.seeMoreModalShown = true;
+        this.modalValueOptions = [...this.filterableValues] as SearchResultModel[];
+    }
+
     @task
     @waitFor
     async applySelectedProperty() {
         if (this.selectedProperty) {
-            const { toggleFilter, propertyCard } = this.args;
-            const card = await this.selectedProperty.indexCard;
+            const { toggleFilter, property } = this.args;
+            const card = this.selectedProperty.indexCard;
             const filter = {
-                property: this.getLocalizedString.compute([propertyCard.get('resourceMetadata'), 'label']),
-                value: this.getLocalizedString.compute([card.resourceMetadata, 'title']),
+                propertyVisibleLabel: property.displayLabel,
+                propertyShortFormLabel: property.shortFormLabel,
+                label: card.get('label'),
+                value: card.get('resourceId'),
             };
             toggleFilter(filter);
             this.selectedProperty = null;
         }
     }
 
+    @task({ restartable: true })
+    @waitFor
+    async debouncedValueSearch(filterString: string) {
+        await timeout(searchDebounceTime);
+        this.filterString = filterString;
+        this.page = '';
+        this.modalValueOptions = [];
+        await taskFor(this.fetchFacetValues).perform();
+    }
+
+    @task
+    @waitFor
+    async loadMoreValues() {
+        this.page = this.nextPageCursor;
+        await taskFor(this.fetchFacetValues).perform();
+    }
+
     @task
     @waitFor
     async fetchFacetValues() {
-        const { cardSearchText, cardSearchFilters } = this.args;
-        const { page, sort } = this;
+        const { cardSearchText, cardSearchFilter, property } = this.args;
+        const { page, sort, filterString } = this;
+        if (booleanFilterProperties.includes(property.shortFormLabel)) {
+            this.filterableValues = [
+                {
+                    resourceId: 'is-present',
+                    indexCard: {
+                        label: this.intl.t('search.filter-facet.has-resource', { resource: property.displayLabel }),
+                        resourceId: 'is-present',
+                    },
+                },
+                {
+                    resourceId: 'is-absent',
+                    indexCard: {
+                        label: this.intl.t('search.filter-facet.no-resource', { resource: property.displayLabel }),
+                        resourceId: 'is-present',
+                    },
+                },
+            ];
+            return;
+        }
```

Comment:
As someone outside of the search project, I'm not really sure what's going on here. This might could do with some documentation.

Reply:
Oh, yea good call. I wrote this yesterday and it took me a second to remember already 😅